pkg/dal/diff.go

```diff
@@ -93,13 +112,30 @@ func (a *Model) Diff(b *Model) (out ModelDiffSet) {
 		_, ok := aIndex[attrB.Ident]
 		if !ok {
 			out = append(out, &ModelDiff{
-				Type:     AttributeMissing,
-				Original: nil,
-				Asserted: attrB,
+				Type:         AttributeMissing,
+                Modification: AttributeAdded,
+                Original:     nil,
+				Asserted:     attrB,
 			})
 			continue
 		}
 	}
 
 	return
 }
+
+// DiffAsMap calculates the diff between models a and b where a is used as base.
+// The output is a map that represents the differences as ModelDiffSet's
+func (a *Model) DiffAsMap(b *Model) (out ModelDiffMap) {
```

Comment:
Do we really need a map for this?

How about `DiffType(ModelModification)` that only returns compatible changes? Or extending `Diff` fn arguments with `...ModelModification`?

Reply:
You're right. It's not needed. I'll remove it.